pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/forward/NoOpMVForwardIndexReader.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.segment.index.readers.forward;
+
+import java.io.IOException;
+import org.apache.pinot.segment.spi.index.reader.ForwardIndexReader;
+import org.apache.pinot.segment.spi.index.reader.ForwardIndexReaderContext;
+import org.apache.pinot.spi.data.FieldSpec;
+
+
+/**
+ * No-op forward index reader meant to be used by multi-value columns with forward index disabled. This is meant to
+ * provide clean exception handling for operations which need to read forward index when forward index is disabled.
+ */
+public class NoOpMVForwardIndexReader implements ForwardIndexReader<ForwardIndexReaderContext> {
```

Comment:
Suggest not using no-op reader as this is not consistent with how we handle other index types. Instead, make `DataSource.getForwardIndex()` return `null` when the forward index does not exist. When forward index is needed to solve the query, check if the forward index is `null` and throw exception accordingly

Reply:
I feel that will be quite tricky and less clean since that requires policing multiple areas in the code and then throw exception. IIRC, we had considered this option but NO-OP reader looked better / cleaner

Reply:
GROUP BY
Transform
Aggregate
SELECT
SCAN based filter
.....


Whoever accesses forward index now needs to be aware of the non-existence of forward index. The NO-OP reader on the other hand does the exact same thing but fails just slightly later by letting the caller attempting to read forward index and then fail with UNSUP. This is much cleaner imo.

Reply:
We should treat forward index the same as other type of indexes, instead of specializing it and always put a dummy reader for it. Let's say in the future I want to support GROUP BY with inverted index only, my branching will be `if (dataSource,getForwardIndex() == null) ...`, similar to when we choose to use the scan based filter when inverted index doesn't exist.
Conceptually we are able to support all operations with inverted index only, so I don't want to specialize forward index just for the exception message.

Reply:
I am not sure I follow. Throwing exception in multiple places using if checks is special casing / specializing it as opposed to using NO-OP reader like done in this PR. We can discuss this more

Reply:
Unless there is a way to do what you are suggesting without leaking it in several different places in the code, I don't think it is a good idea.

Reply:
+1 to what @siddharthteotia says about this. There are too many places which will require null checks and even then we might miss some. New code will also need to deal with such null checks.

Reply:
Not really. We have less than 10 usages of `DataSource.getForwardIndex()` in the production code, so it should be easy to track each of them and ensure the no-forward-index case is properly handled. I don't like the idea of handling forward index differently from other index types - all other index types return `null` when the index does not exist.

Reply:
@siddharthteotia More thoughts on this? Adding a dummy reader can simplify the implementation for now, but in the future we will very likely need to change it to know whether the forward index exist (e.g when supporting re-generating forward index from inverted index). One way to do it is adding it to the data source metadata, but that will make the index handling inconsistent.

Reply:
we discussed and agreed to remove the no-op reader. I've made these changes. thanks for your patience to discuss this @Jackie-Jiang and @siddharthteotia !